src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+public abstract class Contact{
+	
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate) {
+    	this.value = value;
+    	this.isPrivate = isPrivate;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```

Comment:
ouch! you see that you are comparing parent class with one other child class instance? think how you would alleviate this problem?